docs/syntax-attribute-functions-native.md

```diff
@@ -0,0 +1,54 @@
+# Native Attribute for Functions
+
+## Summary
+
+This RFC proposes a function-level `@native` attribute to request native compilation for individual functions, independent of the script-level `--!native` comment directive.
+
+## Motivation
+
+Luau's native compiler currently compiles whole scripts annotated with `--!native` comment directive. However, this provides very coarse-grained control. Since all functions in the script may not benefit from native compilation, developers might be forced to move unrelated functions together to natively compiled scripts. In this RFC, we propose a function-level `@native` attribute to facilitate developers to pick and choose individual functions for native compilation.
```

Comment:
I'm not actually seeing a direct benefit for higher granularity of native codegen. This just says "either the module is native or it isn't" as opposed to benefits for having certain set of functions running native code and others aren't.

Reply:
The part that is being left unsaid is that we have an upper limit to how much code we can compile (the details are not yet final).  Compiling the whole module, including functions that don't benefit from native compilation, unnecessarily exhausts this limit. Function-level granularity can help address this problem.

Reply:
Eventually,  our compiler should automatically decide which functions to compile natively. This might make the `--!native` directive redundant, but the `@native` attribute may still remain useful for creators to explicitly ask for native compilation of specific functions.

Reply:
That part that's being left unsaid should probably be added into the motivation section. That's what it's for! You can leave _specific_ details out, but just give an idea.